devp2p/src/main/kotlin/org/apache/tuweni/devp2p/v5/internal/DefaultNodeDiscoveryService.kt

```diff
@@ -0,0 +1,55 @@
+package org.apache.tuweni.devp2p.v5.internal
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import org.apache.tuweni.bytes.Bytes
+import org.apache.tuweni.crypto.Hash
+import org.apache.tuweni.crypto.SECP256K1
+import org.apache.tuweni.devp2p.EthereumNodeRecord
+import org.apache.tuweni.devp2p.parseEnodeUri
+import org.apache.tuweni.devp2p.v5.NodeDiscoveryService
+import org.apache.tuweni.devp2p.v5.UdpConnector
+import org.apache.tuweni.devp2p.v5.packet.RandomMessage
+import org.apache.tuweni.io.Base64URLSafe
+import java.net.InetSocketAddress
+import java.time.Instant
+import kotlin.coroutines.CoroutineContext
+
+class DefaultNodeDiscoveryService(
+  private val keyPair: SECP256K1.KeyPair,
```

Comment:
Why do we hardcode a SECP256K1 key pair for our own ENR? ENRs take any - consider taking an ENR as parameter instead.

Reply:
Key pair is not for ENR dedicated. There is a private key required for secret agreement